gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1838,7 +1838,14 @@ protected GenerateDataTemplateTask configureDataTemplateGeneration(Project proje
 
     // include additional dependencies into the appropriate configuration used to compile the input source set
     // must include the generated data template classes and their dependencies the configuration
-    String compileConfigName = isTestSourceSet(sourceSet) ? "testCompile" : "compile";
+    String compileConfigName;
+    if (isAtLeastGradle7()) {
```

Comment:
Leave a comment here explaining the context for this check, and maybe add a TODO stating that we can remove this once the min required version (this is defined as a constant near the top of this file) is 7+

Reply:
add a comment